### PR TITLE
improved logs

### PR DESCRIPTION
**Pull request by daminichopra**
Improved logs for [Issue](https://github.com/smartsdk/ngsi-timeseries-api/issues/342)

### CODE REVIEW

File: src/reporter/reporter.py [hunk 1]
```diff
@@ -192,7 +192,7 @@ def notify():
         msg = "Notification not processed or not updated"
```
Comment (chicco785) [inline src/reporter/reporter.py:192]: can we log with error level the payload?

**Review by chicco785 — CHANGES_REQUESTED** [review]

File: src/reporter/reporter.py (hunk 1 above)
Comment (daminichopra) [reply src/reporter/reporter.py:192]: > can we log with error level the payload?

 @chicco785 , As per my understanding we need to add `payload` with error msg `"Notification not processed or not updated"`. Please confirm.

**Comment by chicco785** [comment]
correct!

**Comment by c0c0n3** [comment]
Also if possible, it'd be great if you could add process ID and thread ID to the log line since we're in the process of switching to Gunicorn with a pre-forked runtime. When using async IO, lines get output out of sequence w/r/t code flow. Here's an example of that straight from the Gunicorn logs of the server I'm busy testing

```
[2020-09-07 10:43:37 +0000] [1] [DEBUG] Current configuration:
  config: gunicorn.conf.py
  bind: ['0.0.0.0:8668']
  backlog: 2048
  workers: 9
  worker_class: gevent
  threads: 1
  ...
  worker_connections: 10000
  access_log_format: %(h)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"
  ...
[2020-09-07 10:43:37 +0000] [1] [INFO] Starting gunicorn 20.0.4
[2020-09-07 10:43:37 +0000] [1] [DEBUG] Arbiter booted
[2020-09-07 10:43:37 +0000] [1] [INFO] Listening at: http://0.0.0.0:8668 (1)
[2020-09-07 10:43:37 +0000] [1] [INFO] Using worker: gevent
[2020-09-07 10:43:37 +0000] [8] [INFO] Booting worker with pid: 8
[2020-09-07 10:43:37 +0000] [9] [INFO] Booting worker with pid: 9
[2020-09-07 10:43:37 +0000] [10] [INFO] Booting worker with pid: 10
[2020-09-07 10:43:37 +0000] [11] [INFO] Booting worker with pid: 11
[2020-09-07 10:43:37 +0000] [13] [INFO] Booting worker with pid: 13
[2020-09-07 10:43:37 +0000] [12] [INFO] Booting worker with pid: 12
[2020-09-07 10:43:37 +0000] [14] [INFO] Booting worker with pid: 14
[2020-09-07 10:43:37 +0000] [15] [INFO] Booting worker with pid: 15
[2020-09-07 10:43:37 +0000] [16] [INFO] Booting worker with pid: 16
[2020-09-07 10:43:37 +0000] [1] [DEBUG] 9 workers
[2020-09-07 10:43:56 +0000] [9] [DEBUG] POST /v2/notify
Cannot add location to entity (type: "Room", id: "Room:1"), missing "address" attribute.
2020-09-07 10:43:56.276 DEBUG:reporter.reporter:Received payload: {'id': 'Room:1', 'type': 'Room', 'temperature': {'value': 23, 'type': 'Float'}, 'pressure': {'value': 720, 'type': 'Integer'}, 'time_index': '2020-09-07T10:43:56.228429'}
2020-09-07 10:43:56.283 DEBUG:translators.factory:Env variable QL_CONFIG not set, using default value of: 
2020-09-07 10:43:56.286 DEBUG:translators.factory:no config file specified, using defaults.
2020-09-07 10:43:56.288 INFO:translators.factory:Env variable QL_DEFAULT_DB not set, using default value of: crate
2020-09-07 10:43:56.292 INFO:translators.factory:Backend selected for tenant 'None' is: crate
2020-09-07 10:43:56.299 INFO:translators.crate:Env variable CRATE_HOST set to '192.0.0.1', using this value.
2020-09-07 10:43:56.300 INFO:translators.crate:Env variable CRATE_PORT not set, using default value of: 4200
[2020-09-07 10:43:56 +0000] [9] [DEBUG] POST /v2/notify
2020-09-07 10:43:56.302 DEBUG:connexion.apis.flask_api:Getting data and status code
2020-09-07 10:43:56.302 DEBUG:connexion.decorators.validation:http://192.0.0.1:8668/v2/notify validating schema...
2020-09-07 10:43:56.304 DEBUG:connexion.decorators.validation:http://192.0.0.1:8668/v2/notify validating parameters...
2020-09-07 10:43:56.309 DEBUG:connexion.decorators.parameter:Function Arguments: []
2020-09-07 10:43:56.313 DEBUG:connexion.operations.swagger2:Body parameter 'notification' not in function arguments
```

As you can see, Gunicorn has spawned 9 worker processes and when logging, it includes the PID on the log line but QL statements don't so it becomes very difficult to tie log lines back to code flow, especially in the presence of async IO where lines might get output in a sequence that's different from the sequence of log statements one in the code.

**Comment by daminichopra** [comment]
> Also if possible, it'd be great if you could add process ID and thread ID to the log line since we're in the process of switching to Gunicorn with a pre-forked runtime. When using async IO, lines get output out of sequence w/r/t code flow. Here's an example of that straight from the Gunicorn logs of the server I'm busy testing
> 
> ```
> [2020-09-07 10:43:37 +0000] [1] [DEBUG] Current configuration:
>   config: gunicorn.conf.py
>   bind: ['0.0.0.0:8668']
>   backlog: 2048
>   workers: 9
>   worker_class: gevent
>   threads: 1
>   ...
>   worker_connections: 10000
>   access_log_format: %(h)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"
>   ...
> [2020-09-07 10:43:37 +0000] [1] [INFO] Starting gunicorn 20.0.4
> [2020-09-07 10:43:37 +0000] [1] [DEBUG] Arbiter booted
> [2020-09-07 10:43:37 +0000] [1] [INFO] Listening at: http://0.0.0.0:8668 (1)
> [2020-09-07 10:43:37 +0000] [1] [INFO] Using worker: gevent
> [2020-09-07 10:43:37 +0000] [8] [INFO] Booting worker with pid: 8
> [2020-09-07 10:43:37 +0000] [9] [INFO] Booting worker with pid: 9
> [2020-09-07 10:43:37 +0000] [10] [INFO] Booting worker with pid: 10
> [2020-09-07 10:43:37 +0000] [11] [INFO] Booting worker with pid: 11
> [2020-09-07 10:43:37 +0000] [13] [INFO] Booting worker with pid: 13
> [2020-09-07 10:43:37 +0000] [12] [INFO] Booting worker with pid: 12
> [2020-09-07 10:43:37 +0000] [14] [INFO] Booting worker with pid: 14
> [2020-09-07 10:43:37 +0000] [15] [INFO] Booting worker with pid: 15
> [2020-09-07 10:43:37 +0000] [16] [INFO] Booting worker with pid: 16
> [2020-09-07 10:43:37 +0000] [1] [DEBUG] 9 workers
> [2020-09-07 10:43:56 +0000] [9] [DEBUG] POST /v2/notify
> Cannot add location to entity (type: "Room", id: "Room:1"), missing "address" attribute.
> 2020-09-07 10:43:56.276 DEBUG:reporter.reporter:Received payload: {'id': 'Room:1', 'type': 'Room', 'temperature': {'value': 23, 'type': 'Float'}, 'pressure': {'value': 720, 'type': 'Integer'}, 'time_index': '2020-09-07T10:43:56.228429'}
> 2020-09-07 10:43:56.283 DEBUG:translators.factory:Env variable QL_CONFIG not set, using default value of: 
> 2020-09-07 10:43:56.286 DEBUG:translators.factory:no config file specified, using defaults.
> 2020-09-07 10:43:56.288 INFO:translators.factory:Env variable QL_DEFAULT_DB not set, using default value of: crate
> 2020-09-07 10:43:56.292 INFO:translators.factory:Backend selected for tenant 'None' is: crate
> 2020-09-07 10:43:56.299 INFO:translators.crate:Env variable CRATE_HOST set to '192.0.0.1', using this value.
> 2020-09-07 10:43:56.300 INFO:translators.crate:Env variable CRATE_PORT not set, using default value of: 4200
> [2020-09-07 10:43:56 +0000] [9] [DEBUG] POST /v2/notify
> 2020-09-07 10:43:56.302 DEBUG:connexion.apis.flask_api:Getting data and status code
> 2020-09-07 10:43:56.302 DEBUG:connexion.decorators.validation:http://192.0.0.1:8668/v2/notify validating schema...
> 2020-09-07 10:43:56.304 DEBUG:connexion.decorators.validation:http://192.0.0.1:8668/v2/notify validating parameters...
> 2020-09-07 10:43:56.309 DEBUG:connexion.decorators.parameter:Function Arguments: []
> 2020-09-07 10:43:56.313 DEBUG:connexion.operations.swagger2:Body parameter 'notification' not in function arguments
> ```
> 
> As you can see, Gunicorn has spawned 9 worker processes and when logging, it includes the PID on the log line but QL statements don't so it becomes very difficult to tie log lines back to code flow, especially in the presence of async IO where lines might get output in a sequence that's different from the sequence of log statements in the code. Looking at the above snippet I can't be sure all those lines where actually output by process 9 and async IO (gevent) got them out of sequence or if they actually belong to different processes and there was no async IO shuffling.

@c0c0n3 , Ok I will add process ID and thread ID to the log line but I suggest to please raise a separate ticket for this enhancement if you can. Thank you.

**Comment by c0c0n3** [comment]
@daminichopra 

> Ok I will add process ID and thread ID

you're a superstar, thank you soooo much!

> I suggest to please raise a separate ticket

Sure, here it is #356 :-)

**Comment by daminichopra** [comment]
Hi @chicco785 @c0c0n3 , Please merge and close this PR if it looks good. Thank you

**Review by c0c0n3 — APPROVED** [review]

**Comment by c0c0n3** [comment]
@daminichopra thanks for your PR. At the moment we're having some issues with Travis which is why the build log isn't showing on this page. Here's a link to the latest Travis build for your PR

- https://travis-ci.org/github/smartsdk/ngsi-timeseries-api/builds/726970500

As you can see all the four test sessions failed. That's the bad news. The good news is that I don't think this can possibly have anything to do with your PR, rather something that was already in `master` and you brought in with ed52998. It looks like the Gunicorn server in the docker container suddenly comes to the foreground, so tests stop, and eventually Travis kills the build. Oh dear. I'm struggling to reproduce this scenario on my machine, so am wondering if the **whole** test suite worked for you too locally, i.e. did you run

    $ cd where/you/cloned/ngsi-timeseries-api
    $ pipenv install
    $ source setup_dev_env.sh
    $ sh run_tests.sh

and what was the outcome for you? Did the tests get stuck?! Thanks!!

**Comment by c0c0n3** [comment]
@chicco785 so I've managed to reproduce the issue with a build based on the code currently in `master`:

- https://travis-ci.org/github/smartsdk/ngsi-timeseries-api/builds/727051434

Also, after cleaning up my dev env (docker, python, etc.) I've finally managed to reproduce the issue on my laptop too. Since the issue has definitely nothing to do with this PR, if you agree I'd go ahead and merge this PR into `master`. I opened a separate GitHub issue to tackle the problem: #362.

**Comment by daminichopra** [comment]
> @chicco785 so I've managed to reproduce the issue with a build based on the code currently in `master`:
> 
> * https://travis-ci.org/github/smartsdk/ngsi-timeseries-api/builds/727051434
> 
> Also, after cleaning up my dev env (docker, python, etc.) I've finally managed to reproduce the issue on my laptop too. Since the issue has definitely nothing to do with this PR, if you agree I'd go ahead and merge this PR into `master`. I opened a separate GitHub issue to tackle the problem: #362.

@chicco785 @c0c0n3 , All test cases are passing successfully in my local environment. Please merge  if this issue has nothing to do with this PR . Thanks

**Comment by c0c0n3** [comment]
@daminichopra 

> All test cases are passing successfully in my local environment

excellent! thanks so much for checking. going to merge just now...

**Comment by c0c0n3** [comment]
closes #342

**Pull request merged**